presto-main/src/main/java/com/facebook/presto/memory/MemoryPool.java

```diff
@@ -145,13 +145,21 @@ public void removeTaskRevocableMemoryListener(TaskRevocableMemoryListener listen
             }
         }
 
-        onMemoryReserved();
+        onMemoryReserved(queryId);
         return result;
     }
 
-    private void onMemoryReserved()
+    private void onMemoryReserved(QueryId queryId)
     {
-        listeners.forEach(listener -> listener.onMemoryReserved(this));
+        listeners.forEach(listener -> listener.onMemoryReserved(
+                this,
+                queryId,
+                valueOrZero(queryMemoryReservations.get(queryId)) + valueOrZero(queryMemoryRevocableReservations.get(queryId))));
```

Comment:
Not a big deal since we only expect on listener to be registered, but to be safe you probably want to hoist the memory reservation computation out of the loop.

Reply:
Also, why not `queryMemoryReservations.getOrDefault(queryId, 0L)` ?